### PR TITLE
Disallow creating a repository with a .git suffix

### LINKED ISSUE #956: Remove ".git" suffix when creating a repository
To simplify the operation that uses the repository name such as the authorization system.
GitHub also removes the suffix automatically when creating a repository.
image

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
It is undesirable for repositories to be created with a `.git` suffix due to potential conflicts or confusion with Git's repository structure and management.

Modifications:
- Disallow creating a repository whose suffix is `.git`

Result:
- You can no longer create a repository whose suffix is `.git`
- Close #956

### CODE REVIEW

**Review by ikhoon — APPROVED** [review]
👍

File: server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java [hunk 1]
```diff
@@ -437,7 +438,8 @@ private static boolean isValidChildName(String name) {
             return false;
         }
 
-        return !name.endsWith(SUFFIX_REMOVED) && !name.endsWith(SUFFIX_PURGED);
+        return !name.endsWith(SUFFIX_REMOVED) && !name.endsWith(SUFFIX_PURGED) &&
+               !name.endsWith(GIT_EXTENSION); // Git repository whose name ends with '.git' is not allowed.
```
Comment (jrhee17) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java:442]: Note: I understood if there are any users out there who use `.git` suffixes, the project/repositories just won't be loaded server start-up will be successful.

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

File: server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java (hunk 1 above)
Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java:442]: That was my intention. 👍

**Pull request merged**